floss/language/go/extract.py

```diff
@@ -0,0 +1,431 @@
+        except UnicodeDecodeError:
+            continue
+
+        if section_name == ".text":
+            # Extract go Build ID
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            # Build ID is a string that starts with "\xff\x20 Go build ID: " and ends with "\n"
+            # FF 20 47 6F 20 62 75 69  6C 64 20 49 44 3A 20 22  . Go build ID: "
+            # 36 4E 31 4D 77 6E 30 31  72 46 6E 41 51 4B 62 5A  6N1Mwn01rFnAQKbZ
+            # 73 46 5A 32 2F 38 41 6B  75 63 4B 46 58 4D 49 54  sFZ2/8AkucKFXMIT
+            # 63 51 52 49 75 55 5F 79  32 2F 62 76 4F 67 56 37  cQRIuU_y2/bvOgV7
+            # 52 4D 54 72 77 73 7A 5A  39 57 7A 69 6C 64 2F 72  RMTrwszZ9Wzild/r
+            # 33 31 50 47 70 61 6B 2D  48 77 36 4B 72 77 59 6E  31PGpak-Hw6KrwYn
+            # 52 4E 73 22 0A 20 FF CC  CC CC CC CC CC CC CC CC  RNs". ..........
+
+            build_id_regex = re.compile(b"(?<=\xff\x20)(.)*\x0A")
+
+            for m in build_id_regex.finditer(section_data):
+                addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                try:
+                    string = m.group(0).decode("utf-8")
+                    yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                except UnicodeDecodeError:
+                    continue
+                break
+
+        if section_name == ".text":
+            # Extract long strings
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            if alignment == 0x10:
+                for m in longstring64.finditer(section_data):
+                    format = "<I"
+                    s_off = struct.unpack(format, m.group(1))[0]
+                    s_size = struct.unpack(format, m.group(2))[0]
+
+                    s_rva = s_off + m.start() + section_va + 7
+                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                    try:
+                        string = pe.get_string_at_rva(s_rva, s_size).decode("utf-8")
+                        if string.isprintable() and string != "" and len(string) >= min_length:
+                            yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                    except UnicodeDecodeError:
+                        continue
+
+                for m in longstring64_2.finditer(section_data):
+                    s_off = struct.unpack("<I", m.group(3))[0]
+                    s_size = struct.unpack("<B", m.group(1))[0]
+
+                    s_rva = s_off + m.end() + section_va
+                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                    try:
+                        string = pe.get_string_at_rva(s_rva, s_size).decode("utf-8")
```

Comment:
side bar on performance: as this appears to be slow we could just get the data from the section_data directly using s_off